secator/tasks/shodan.py

```
@@ -0,0 +1,163 @@
+from secator.decorators import task
+from secator.definitions import HOST, IP
+from secator.output_types import (
+	Error, Ip, Port, Subdomain, Tag, Technology, Vulnerability, Warning
+)
+from secator.runners import PythonRunner
+
+
+@task()
+class shodan(PythonRunner):
+	"""Passive host recon via the Shodan API (ports, services, CVEs, hostnames)."""
+	input_types = [HOST, IP]
```

Ruff (0.15.20)

[warning] Mutable default value for class attribute

(RUF012)